demo/src/components/Product/Product.js

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import {Link} from 'react-router-dom';
+import {Image} from 'react-bootstrap';
+import PropTypes from 'prop-types';
+import './Product.css';
+
+/**
+ * Creates a display component for a single product.
+ * @param {string} id
+ * @param {string} title
+ * @param {string} price
+ * @param {string} imageSrc
+ * @param {string} description
+ * @return {!JSX} The component.
+ */
+function Product({id, title, price, imageSrc, description}) {
```

Comment:
The name "Product" on its own feels overly broad.  For widgets, I usually prefer a suffix indicating what type of widget it is or where it will be used.

I'd probably use a name like ProductOverview or ProductListing or something to highlight that it's a general-purpose way of displaying just the most information about a product and avoid the ambiguity.  If there's a standard React name for this concept, then use that instead.

This falls into the style guide [guidance](https://google.github.io/styleguide/jsguide.html#naming-rules-common-to-all-identifiers) to "as descriptive a name as possible, within reason."